chore(docs): keys concept page refactor

signorecello opened the pull request:
# Reposition Accounts Documentation and Enhance Key Management Documentation

This PR makes two key improvements to the developer documentation:

1. Repositions the Accounts section to the top of the concepts hierarchy by changing its position from 5 to 0, and removes an unnecessary sidebar position parameter.

2. Completely revamps the Keys documentation with a more comprehensive and structured explanation:
   - Adds detailed explanations of the four key types (nullifier, address, incoming viewing, signing)
   - Introduces comparison tables for easier understanding
   - Adds a mermaid diagram showing the encryption flow
   - Expands key storage patterns with pros/cons analysis
   - Adds sections on key management, app-siloing, and rotation
   - Includes advanced use cases like escrow contracts
   - Improves code examples and visual organization

These changes make the documentation more accessible to developers while providing deeper technical insights into Aztec's key management system.

signorecello commented:
* **#17509** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17509?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17509?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#17185** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17185?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17161** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17161?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

critesjosh commented on docs/docs/developers/docs/concepts/accounts/keys.md:
```diff
@@ -1,155 +1,307 @@
+pre_address = hash(public_keys_hash, partial_address)
+
+where:
+  public_keys_hash = hash(Npk_m, Ivpk_m, Ovpk_m, Tpk_m)
```
We haven't mentioned what Ivpk_m, Ovpk_m. Based on the diagram, i removed `Tpk_m`

critesjosh replied:
maybe move the "Address Keys" section to to after the section on incoming keys

critesjosh commented on docs/docs/developers/docs/concepts/accounts/keys.md:
````diff
@@ -1,155 +1,307 @@
-- The recipient gets a pair (`Epk`, `Ciphertext`)
-- The recipient uses the `address_sk` to decrypt the secret: `S = Epk * address_sk`.
-- The recipient uses the decrypted secret to decrypt the ciphertext.
+```mermaid
````
This diagram looks really small on my laptop screen. maybe break it up into two rows?

<img width="2161" height="1091" alt="image" src="https://github.com/user-attachments/assets/6f26a9fd-fdee-4192-b765-0d974ee46688" />

critesjosh commented on docs/docs/developers/docs/concepts/accounts/keys.md:
```diff
@@ -1,155 +1,307 @@
+**Purpose**: Receiving and decrypting private notes
 
-When it comes to notes encryption and decryption:
+Incoming viewing keys enable private transfer of value and data. The sender uses the recipient's public viewing key (`Ivpk`) to encrypt notes, and the recipient uses their secret viewing key (`ivsk`) to decrypt them.
```
```suggestion
Incoming viewing keys enable private information to be shared with recipients. The sender uses the recipient's public viewing key (`Ivpk`) to encrypt notes, and the recipient uses their secret viewing key (`ivsk`) to decrypt them.
```

critesjosh commented on docs/docs/developers/docs/concepts/accounts/keys.md:
```diff
@@ -1,155 +1,307 @@
+- Secure key storage and oracle access
+- Key material never leaves the client
+
+All keys use elliptic curve cryptography on the [Grumpkin curve](https://github.com/AztecProtocol/aztec-connect/blob/9374aae687ec5ea01adeb651e7b9ab0d69a1b33b/markdown/specs/aztec-connect/src/primitives.md):
```
this links to the Aztec Connect  repo
```suggestion
All keys use elliptic curve cryptography on the Grumpkin curve:
```

critesjosh commented on docs/docs/developers/docs/concepts/accounts/keys.md:
```diff
@@ -1,155 +1,307 @@
+- **Reduces account security** - compromises separation of concerns
+- Not recommended for production use
+
+## Key Management
```
I think this section is more important than Signing Key storage patterns. lets move this section up

critesjosh commented on docs/docs/developers/docs/concepts/accounts/keys.md:
```diff
@@ -1,155 +1,307 @@
+### Non-Account Contracts with Keys
 
-An exception (a non-account contract which would have some of the keys non-zero) is an escrow contract. Escrow contract is a type of contract which on its own is an "owner" of a note meaning that it has a `Npk_m` registered and the notes contain this `Npk_m`.
+Typically, only account contracts have non-zero public keys. However, some contracts like **escrow** or **custodial** contracts need keys even though they aren't accounts.
```
```suggestion
Typically, only account contracts have non-default public keys. However, some contracts like **escrow** or **custodial** contracts need keys even though they aren't accounts.
```
Default public keys are not zero, see here: https://github.com/AztecProtocol/aztec-packages/blob/e82340416e2237d30e64cc0034d44eed895ec514/yarn-project/stdlib/src/keys/public_keys.ts#L78

critesjosh reviewed (changes requested):
Overall, looks good, just some minor suggestions. I made some edit after telling claude to review this PR, following the conventions outlined in docs/CLAUDE.md

critesjosh commented on docs/docs/developers/docs/concepts/accounts/keys.md:
```diff
@@ -1,155 +1,307 @@
+- Each transaction nullifies and recreates the note (expensive)
+- Higher cost per transaction
+
+### Delayed Public Mutable State
```
I know this is possible, but do we want to recommend it as an option? what is a case where having the public key be public be useful?

critesjosh commented on docs/docs/developers/docs/concepts/accounts/keys.md:
```diff
@@ -1,155 +1,307 @@
+- Higher proving time per transaction
+- More complex architecture
+
+### Reuse Protocol Keys (Not Recommended)
```
since this is not recommended, let's remove it

signorecello committed:
chore(docs): keys concept page refactor

# Reposition Accounts Documentation and Enhance Key Management Documentation

This PR makes two key improvements to the developer documentation:

1. Repositions the Accounts section to the top of the concepts hierarchy by changing its position from 5 to 0, and removes an unnecessary sidebar position parameter.

2. Completely revamps the Keys documentation with a more comprehensive and structured explanation:
   - Adds detailed explanations of the four key types (nullifier, address, incoming viewing, signing)
   - Introduces comparison tables for easier understanding
   - Adds a mermaid diagram showing the encryption flow
   - Expands key storage patterns with pros/cons analysis
   - Adds sections on key management, app-siloing, and rotation
   - Includes advanced use cases like escrow contracts
   - Improves code examples and visual organization

These changes make the documentation more accessible to developers while providing deeper technical insights into Aztec's key management system.
